service/dap/server.go

```diff
@@ -836,6 +836,18 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		}
 	}
 
+	backend, ok := request.Arguments["backend"]
```

Comment:
is it simpler if use

backend, ok := request.Arguments["backend"].(string)
if !ok {
   ...
}

I think ok will be false if backend attribute doesn't exist so request.Arguments["backend"] is nil.

Reply:
We combine the checks like this for required args, but split them for optional ones. This way if the arg is not specified, it won't be an error and a default will be used instead. (In this case, the default is supplied in command.go, but I can make it explicit in the server like with output). And if the arg is specified, then we only accept valid values and complain about anything malformed. This code is getting quite repetitive. I have been playing with refactoring and helpers in another PR.

Reply:
fyi - for config parsing I am proposing a different approach in #2571.

Reply:
I have updated the code to make it clear that missing value is not an error and a default is used instead. 
I don't think it is worth massaging this code to be more succinct since you have a PR that impacts this code anyway.